src/obstacle_params_parser.cpp

```diff
@@ -0,0 +1,133 @@
+#pragma once
+
+#include "panda_torque_mpc/obstacle_params_parser.h"
+
+#include <Eigen/Dense>
+#include <boost/shared_ptr.hpp>
+#include <fstream>
+#include <memory>
+#include <vector>
+
+#include <pinocchio/fwd.hpp>
+#include <pinocchio/multibody/model.hpp>
+
+#include <ros/ros.h>
+#include <yaml-cpp/yaml.h>
+
+namespace panda_torque_mpc {
+
+  ObstacleParamsParser::ObstacleParamsParser(
+      const std::shared_ptr<ros::NodeHandle> &nh,
+      const boost::shared_ptr<pinocchio::Model> &pin_model,
+      const boost::shared_ptr<pinocchio::GeometryModel> &collision_model)
+      : nh_(nh), pin_model_(pin_model), collision_model_(collision_model){};
+
+  ObstacleParamsParser::~ObstacleParamsParser() {};
+
+void ObstacleParamsParser::addCollisions()
+{
+  // First adding the obstacles to the geometry model
+ int obs_idx = 0;
+ while (nh_->hasParam("obstacle_" + std::to_string(obs_idx))) {
+
+  std::string obstacle_name;
+
+  obstacle_name = "obstacle_" + std::to_string(obs_idx);
+  std::string type;
+
+  nh_->getParam(obstacle_name + "/type", type);
+
+  Eigen::VectorXd pose;
+  nh_->getParam(obstacle_name + "/pose", pose);
```

Comment:
The way you did it unfortunately will not work. This constructor expects `std::vector<double>`.
You can obtain result you want this way:
```suggestion
  std::vector<double> pose_vect;
  nh_->getParam(obstacle_name + "/pose", pose_vect);
  // check if pose_vect.size() == 3 and print error if not
  const auto translation = Eigen::Map<Eigen::Vector3d>(pose_vect.data()) ;
```
This is only suggestion, and I am not 100% if this will compile. Yet, this is the way you should solve this problem.

Additionally, this removes the need to later transform the `pose` of a type `VectorXd` into `Vector3d`.

Reply:
Let's get rid of this pose and separate the pose in rotation and translation part, which are respectively `Vector3d` and `Vector4d`

Reply:
Good idea. Current implementation also looks good